dev_tools/nyaml2nxdl/nyaml2nxdl_forward_tools.py

```diff
@@ -0,0 +1,1242 @@
+            obj.set("optional", "true")
+        elif value == "recommended":
+            obj.set("recommended", "true")
+        elif value == "required":
+            obj.set("optional", "false")
+        else:
+            obj.set("minOccurs", "0")
+
+
+# pylint: disable=too-many-branches, too-many-locals, too-many-statements
+def xml_handle_group(dct, obj, keyword, value, verbose=False):
+    """
+    The function deals with group instances
+    """
+    line_number = f"__line__{keyword}"
+    line_loc = dct[line_number]
+    xml_handle_comment(obj, line_number, line_loc)
+    list_of_attr = [
+        "name",
+        "type",
+        "nameType",
+        "deprecated",
+        "optional",
+        "recommended",
+        "exists",
+        "unit",
+    ]
+    l_bracket = -1
+    r_bracket = -1
+    if keyword.count("(") == 1:
+        l_bracket = keyword.index("(")
+    if keyword.count(")") == 1:
+        r_bracket = keyword.index(")")
+
+    keyword_name, keyword_type = nx_name_type_resolving(keyword)
+    if not keyword_name and not keyword_type:
+        raise ValueError("A group must have both value and name. Check for group.")
+    grp = ET.SubElement(obj, "group")
+
+    if l_bracket == 0 and r_bracket > 0:
+        grp.set("type", keyword_type)
+        if keyword_name:
+            grp.set("name", keyword_name)
+    elif l_bracket > 0:
+        grp.set("name", keyword_name)
+        if keyword_type:
+            grp.set("type", keyword_type)
+    else:
+        grp.set("name", keyword_name)
+
+    if value:
+        rm_key_list = []
+        for attr, vval in value.items():
+            if "__line__" in attr:
+                continue
+            line_number = f"__line__{attr}"
+            line_loc = value[line_number]
+            if attr == "doc":
+                xml_handle_doc(grp, vval, line_number, line_loc)
+                rm_key_list.append(attr)
+                rm_key_list.append(line_number)
+            elif attr == "exists" and vval:
+                xml_handle_exists(value, grp, attr, vval)
+                rm_key_list.append(attr)
+                rm_key_list.append(line_number)
+                xml_handle_comment(obj, line_number, line_loc, grp)
+            elif attr == "unit":
+                xml_handle_units(grp, vval)
+                xml_handle_comment(obj, line_number, line_loc, grp)
+            elif attr in list_of_attr and not isinstance(vval, dict) and vval:
+                validate_field_attribute_and_value(attr, vval, list_of_attr, value)
+                grp.set(attr, check_for_mapping_char_other(vval))
+                rm_key_list.append(attr)
+                rm_key_list.append(line_number)
+                xml_handle_comment(obj, line_number, line_loc, grp)
+
+        for key in rm_key_list:
+            del value[key]
+        # Check for skipped attrinutes
+        check_for_skiped_attributes("group", value, list_of_attr, verbose)
+    if isinstance(value, dict) and value != {}:
```

Comment:
value is assumed to be a dict so just check its length. Same issue in code below.

Reply:
:+1: